lib/cheetah.rb

```diff
@@ -375,6 +378,7 @@ class << self
 
     def run(*args)
       options = args.last.is_a?(Hash) ? args.pop : {}
+      return changed_env(options, *args) if options[:env]
```

Comment:
This implementation using `return` here to do a hidden recursive call to `run` is far from being obvious. I think it's better to do things more explicit.